DragonSlayer.h

```diff
@@ -1,8 +1,17 @@
 #pragma once
 
+#include "Character.h"
+#include "AttackItem.h"
+
 struct DragonSlayer : Character
 {
+    DragonSlayer(const std::string& name_, int hp_, int armor_);
+    ~DragonSlayer() override;
+    virtual const std::string& getName() override;
+    virtual std::string getStats() override;
+
     void attack(Character& other) override;
 private:
     const std::string name;
-};
\ No newline at end of file
+    std::unique_ptr<AttackItem> attackItem = nullptr;
```

Comment:
unique_ptr does not need initialization to nullptr.
that's the value it has when it is constructed.
You can initialize it with the `AttackItem` directly using `make_unique` in-class:
```
unique_ptr<..> attackItem = std::make_unique<...>(...);
```
- [ ] revise